rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -1324,4 +1457,146 @@ size_t Player::get_number_of_registered_on_play_msg_post_callbacks()
   return pimpl_->get_number_of_registered_on_play_msg_post_callbacks();
 }
 
+PlayerImpl::PlayerClient::PlayerClient(
+  std::shared_ptr<rclcpp::GenericClient> cli,
+  std::string service_name,
+  std::string service_event_type,
+  const rclcpp::Logger & logger)
+: client_(std::move(cli)),
+  service_name_(service_name),
+  logger_(logger)
+{
+  ts_lib_ = rclcpp::get_typesupport_library(
+    service_event_type, "rosidl_typesupport_cpp");
+
+  ts_ = rclcpp::get_typesupport_handle(
+    service_event_type,
+    "rosidl_typesupport_cpp",
+    *ts_lib_);
+
+  auto ts_handle = get_message_typesupport_handle(
+    ts_,
+    rosidl_typesupport_introspection_cpp::typesupport_identifier);
+
+  message_members_ =
+    reinterpret_cast<const rosidl_typesupport_introspection_cpp::MessageMembers *>(
+    ts_handle->data);
+}
+
+bool PlayerImpl::PlayerClient::is_include_request_message(const rclcpp::SerializedMessage & message)
+{
+  auto type = get_msg_event_type(message);
+
+  // Ignore response message
+  if (type == service_msgs::msg::ServiceEventInfo::RESPONSE_SENT ||
+    type == service_msgs::msg::ServiceEventInfo::RESPONSE_RECEIVED)
+  {
+    return false;
+  }
+
+  // Ignore metadata message
+  if (type == service_msgs::msg::ServiceEventInfo::REQUEST_SENT &&
+    client_side_type_ == introspection_type::METADATA)
+  {
+    return false;
+  }
+  if (type == service_msgs::msg::ServiceEventInfo::REQUEST_RECEIVED &&
+    service_side_type_ == introspection_type::METADATA)
+  {
+    return false;
+  }
```

Comment:
Comments from MichaelOrlov

```suggestion
  // Ignore metadata message
  if ((type == service_msgs::msg::ServiceEventInfo::REQUEST_SENT ||
    type == service_msgs::msg::ServiceEventInfo::REQUEST_RECEIVED) &&
    client_side_type_ == introspection_type::METADATA)
  {
    return false;
  }
```

We can simplify these two conditions by combining them in one.

Reply:
Updated.